src/snps/io/reader.py

```diff
@@ -793,27 +798,65 @@ def parser():
             data_io = io.StringIO(data)
             df = pd.read_csv(data_io, sep="\t", dtype=dtypes, na_values=na_values)
 
-            def map_rsids(x):
-                return gsa_resources["rsid_map"].get(x)
+            rev_comp = {}
+            rev_comp["A"] = "T"
+            rev_comp["T"] = "A"
+            rev_comp["G"] = "C"
+            rev_comp["C"] = "G"
 
-            def map_chr(x):
-                chrpos = gsa_resources["chrpos_map"].get(x)
-                return chrpos.split(":")[0] if chrpos else None
+            def map_row(row):
+                snp_name = row["SNP Name"]
 
-            def map_pos(x):
-                chrpos = gsa_resources["chrpos_map"].get(x)
-                return chrpos.split(":")[1] if chrpos else None
+                row["rsid"] = gsa_resources["rsid_map"].get(snp_name, snp_name)
 
-            df["rsid"] = df["SNP Name"].apply(map_rsids)
-            df["chrom"] = df["SNP Name"].apply(map_chr)
-            df["pos"] = df["SNP Name"].apply(map_pos)
-            df["genotype"] = (
-                df["Allele1 - {}".format(strand)] + df["Allele2 - {}".format(strand)]
-            )
-            df.dropna(subset=["rsid", "chrom", "pos"], inplace=True)
+                # prefer the chromosone and position from the map (because ?)
```

Comment:
Is this a question you want answered? I'm not 100% sure of the answer myself.

Reply:
Also - chromosome*

Reply:
Ha, I think I left it in as a note to myself to come back to - didn't really mean it to be in the codebase. 

It seems this information is in the source file, so changing it without a good reason seems dangerous as it might lead to quite confusing behaviour in future (e.g. 37 vs 38 detection).

Reply:
After a conversion, we decided that we should prefer the existing chromosome/location. In theory, that should always be present but I've kept the fallback to the external location for now.